src/components/Pickers/Lookup/Lookup.js

```diff
@@ -144,8 +145,8 @@ class Lookup extends React.PureComponent {
 		});
 
 		const { optionsCache } = this.state;
-
-		if (!optionsCache[search]) {
+		const shouldLoadOptions = !optionsCache[search] && (search || this.props.allowSearchWithEmptyFilter);
```

Comment:
AlwaysRefresh means that Lookup will always fetch options on menu open instead of taking them from cache. This expression says that on each input change options should be loaded if there's some search text or search without criteria is allowed.